client.go

```diff
@@ -166,3 +178,13 @@ func (c *Client) withLockContext(fn func()) {
 	defer c.Unlock()
 	fn()
 }
+
+// Shutdown stops the infinite renewToken loop and cleans up any associated
+// resources for the client. You can call `defer client.Shutdown()` safely even
```

Comment:
Similar to my comment above.

```suggestion
// resources for the client. You can call defer client.Shutdown() safely even
```